src/whisper.cpp

```diff
@@ -7614,6 +7614,31 @@ int whisper_full_with_state(
                                 result_all.back().tokens.push_back(tokens_cur[j]);
                             }
 
+                            // Check if first segment has skipped audio (multiple timestamp tokens after [_BEG_]) by
+                            // peeking at a few of the tokens following, and see if their timestamps are increasing.
+                            // If this is the case we adjust the first segment's start timestamp (t0) to account for
+                            // the skipped audio.
+                            //
+                            // Note that this will only happen for models that have been trained to skip
+                            // introductions/credits like the medium models for example.
+                            const int n_check_tokens = 6;
+                            const int thold_intro_detect = 300;
+                            auto & segment = result_all[0];
+                            auto & tokens  = segment.tokens;
+                            if (segment.t0 == 0 && tokens.size() > n_check_tokens) {
+                                int sum = 0;
+                                for (int i = 0; i < n_check_tokens; ++i) {
+                                    sum += tokens[i].tid - whisper_token_beg(ctx);
+                                }
+                                // If the total sum of the first 6 timestamp tokens is larger than 300
+                                // centiseconds (3 seconds) then we take it to mean that there is an
+                                // introduction/credits that the model skipped.
+                                if (sum > thold_intro_detect) {
+                                    segment.t0 = state->t_beg + 2*(tokens[2].tid - whisper_token_beg(ctx));
+                                    WHISPER_LOG_INFO("%s: detected skipped audio sequence, setting segment.t0 to %d\n", __func__, (int)segment.t0);
+                                }
+                            }
+
```

Comment:
I think this fix is not necessary. I have doubts that this will work in all cases and also skipping intro of trained audio is not a big problem because we are mainly interested on applying the model on new audio. So let's remove it for now and merge only the timestamp fixes.